kernel/sched.c

```diff
@@ -1497,7 +1497,10 @@ void z_thread_abort(struct k_thread *thread)
 	if (active) {
 		/* It's running somewhere else, flag and poke */
 		thread->base.thread_state |= _THREAD_ABORTING;
+
+#ifdef CONFIG_SCHED_IPI_SUPPORTED
 		arch_sched_ipi();
+#endif /* CONFIG_SCHED_IPI_SUPPORTED */
```

Comment:
Style: that comment looks kinda silly when the matching #if is just two lines above.   Maybe what we really want is a noop fallback defined when !SCHED_IPI_SUPPORTED to remove the need for #ifery.